lib/neo4j/core/cypher_session/adaptors/http.rb

```diff
@@ -94,12 +95,12 @@ class Requestor
             default_url('http://neo4:neo4j@localhost:7474')
             validate_uri { |uri| uri.is_a?(URI::HTTP) }
 
-            def initialize(url, user_agent_string, instrument_proc)
+            def initialize(url, user_agent_string, instrument_proc, adaptor)
               self.url = url
               @user = user
               @password = password
               @user_agent_string = user_agent_string
-              @faraday = faraday_connection
+              @faraday = faraday_connection(adaptor)
```

Comment:
I would agree with you, except that the options you can pass into Faraday.new are the **per-request** options, not the connection adapter options.

    # options - The optional Hash used to configure this Faraday::Connection.
    #           Any of these values will be set on every request made, unless
    #           overridden for a specific request.
    #           :url     - String base URL.
    #           :params  - Hash of URI query unencoded key/value pairs.
    #           :headers - Hash of unencoded HTTP header key/value pairs.
    #           :request - Hash of request options.
    #           :ssl     - Hash of SSL options.

For the adapter options, you have to pass in a block and call methods on the invoked parameter:

    conn = Faraday.new(:url => 'http://sushi.com') do |faraday|
      faraday.request  :url_encoded             # form-encode POST params
      faraday.response :logger                  # log requests to STDOUT
      faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
    end

I guess we could do it like this:

    http_adaptor = Neo4j::Core::CypherSession::Adaptors::HTTP.new('http://neo4j:pass@localhost:7474') do |faraday|
      faraday.request  :url_encoded             # form-encode POST params
      faraday.response :logger                  # log requests to STDOUT
      faraday.adapter  Faraday.default_adapter  # make requests with Net::HTTP
    end

...the only challenge there is that there is a Typhoeus issue forcing you to [`require 'typhoeus/adapters/faraday'`](https://github.com/typhoeus/typhoeus/issues/226)

There's probably a way to make it work, but the current PR should not require anyone to add the typhoeus gem if they don't use it - if you look, I [require other gems in the `Gemfile`](https://github.com/brucek/neo4j-core/blob/ca48b3482bc1e0595ccb07e3e67017d5b519b486/Gemfile#L28) for tests, but not in the `.gemspec`.

The other issue is that - scoping out - when an end Rails user needs to configure the gem, they can ideally set whatever options in neo4j.yml (seems like that is still the preferred method, yes?).  So maybe the answer is to start with faraday_adapter options there and work down?

Looking more into it, we're creating the Faraday connection like so:

    conn = Faraday.new(url, init_params) do |b|
      b.request :basic_auth, params[:basic_auth][:username], params[:basic_auth][:password] if params[:basic_auth]
      b.request :multi_json
      # b.response :logger, ::Logger.new(STDOUT), bodies: true
  
      b.response :multi_json, symbolize_keys: true, content_type: 'application/json'
      # b.use Faraday::Response::RaiseError
      b.adapter http_adaptor
    end

...which makes it a tricky to deal with YAML-ifying the options.  I just updated the structure to what you suggest, but only support the `:adapter` option to `:faraday_options` currently in order to make the work more digestible (to me ;-), perhaps someone else will care enough to do the remaining heavy lifting to support generic Faraday options?